network/transport/v2/handlers.go

```diff
@@ -47,32 +48,66 @@ func (p protocol) Handle(peer transport.Peer, raw interface{}) error {
 	return nil
 }
 
-func (p protocol) handle(peer transport.Peer, envelope *Envelope) error {
-	switch msg := envelope.Message.(type) {
+type handleFunc func(peer transport.Peer, envelope *Envelope) error
+
+func (p *protocol) handleASynchWithLock(peer transport.Peer, envelope *Envelope, f handleFunc) error {
```

Comment:
The problem this fix addresses is that a node can be too busy processing transactions to do anything else. The quickest way for this node to be active again is to complete its task. Doing two or three things at once will probably take longer in total.

If a node is busy processing, other nodes should probably look to other nodes for their new transactions. This would argue for not limiting the number of total conversations (maybe just per peer).

If a node is processing a transactionRangeQuery, it already has a RLock so the Lock has to wait until the lock has been removed.